elasticdl/python/common/k8s_client.py

```diff
@@ -81,16 +82,20 @@ def __init__(
             self.cluster = cluster_spec_module.cluster
 
     def _watch(self):
-        stream = watch.Watch().stream(
-            self.client.list_namespaced_pod,
-            self.namespace,
-            label_selector=ELASTICDL_JOB_KEY + "=" + self.job_name,
-        )
-        for event in stream:
-            try:
-                self._event_cb(event)
-            except Exception:
-                traceback.print_exc()
+        while True:
+            stream = watch.Watch().stream(
+                self.client.list_namespaced_pod,
+                self.namespace,
+                label_selector=ELASTICDL_JOB_KEY + "=" + self.job_name,
+            )
+            for event in stream:
+                try:
+                    self._event_cb(event)
+                except Exception:
+                    traceback.print_exc()
+            # In case of any flaky issue causing exceptions, we wait for little
```

Comment:
Do we need a timeout? Otherwise this will keep running until no exception happens.

Reply:
We can add a max try time here

Reply:
Yes, this thread should run forever, and its lifecycle should depend on the master lifecycle.